src/udiskslinuxpartition.c

```diff
@@ -265,8 +298,10 @@ udisks_linux_partition_update (UDisksLinuxPartition   *partition,
         }
     }
 
-  if (disk_block_object != NULL)
+  if (disk_block_object != NULL) {
     table_object_path = g_dbus_object_get_object_path (G_DBUS_OBJECT (disk_block_object));
+    update_partitions_list (disk_block_object, object);
```

Comment:
Is this running in the main thread? If not I think there's a risk that multiple partitions would be updated at once and the list would get corrupted/unpredictable. In that case we need to make the `update_partitions_list` function thread-safe.

Reply:
I always thought the update functions run only in the main thread, but there is a lock (`provider_lock` in `udiskslinuxprovider.c`) that prevents handling multiple uevents at once, so it is possible that it actually can run from different threads (but the lock is there and the update can run only as a result of an add/change uevent, so it's safe).